subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/FileCacheBackedScriptClassCompiler.java

```diff
@@ -82,7 +82,7 @@ public FileCacheBackedScriptClassCompiler(CacheRepository cacheRepository, Scrip
         }
 
         HashCode sourceHashCode = source.getResource().getContentHash();
-        final String sourceHash = HashUtil.compactStringFor(sourceHashCode);
+        final String sourceHash = HashUtil.compactStringFor(sourceHashCode.toByteArray());
```

Comment:
Since we already have a `fromBytesNoCopy` for cases when we know it's safe, do you think it would make sense to add a `toByteArrayNoCopy` as well?

Reply:
Where would that be useful?

Reply:
Ww don't need to clone the internal byte array for `HashUtil.compactStringFor`, might as well reuse it.

Reply:
Not sure how to make that work in a safe way. In any case I'd like to remove most usages of `compactStringFor()` soon, including all that uses a `HashCode`.